daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Inputs.scala

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.transaction
+
+import com.daml.lf.data.Ref
+import com.daml.lf.value.Value.ContractId
+
+import scala.collection.immutable.TreeSet
+
+/** The inputs of the transaction, that is:
+  *   - the contracts were fetched
+  *   - all the parties involved
+  *   - all the contract keys involved.
+  */
+/** @param contracts
+  *     The contracts fetched by this transaction.
+  *
+  * @param parties
+  *     The contracts created by this transaction.
+  *     When the transaction is committed, keys marking the activeness of these
+  *     contracts should be created. The key should be a combination of the transaction
+  *     id and the relative contract id (that is, the node index).
+  * @param keys
+  *     The contract keys created or updated as part of the transaction.
```

Comment:
This information is not enough to check the preconditions of a transaction w.r.t. the ledger state. For example, it's not clear whether the keys should be unassigned or assigned at the beginning of the transaction.

Reply:
Yes, I need this to.

Reply:
andreaslochbihler-da
> This information is not enough to check the preconditions of a transaction w.r.t. the ledger state. For example, it's not clear > whether the keys should be unassigned or assigned at the beginning of the transaction.

Let's tackle that in an upcoming PR.